Mathlib/Data/Nat/Factorial/NatCast.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir, María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos-Fernández
+-/
+
+import Mathlib.Algebra.Algebra.Defs
+import Mathlib.Algebra.CharP.Defs
+import Mathlib.Algebra.Order.Field.Rat
+import Mathlib.Data.Int.GCD
+import Mathlib.Data.Nat.Factorial.Basic
+import Mathlib.Data.Nat.Prime.Basic
+
+/-!
+# Invertibility of factorials
+
+This file contains lemmas providing sufficient conditions for the cast of `n!` to a commutative
+(semi)ring `A` to be a unit.
+
+-/
+
+section Factorial
+
+open Nat
+
+section Semiring
+
+variable {A : Type*} [CommSemiring A]
+
+theorem natCast_factorial_isUnit_of_lt {n : ℕ} (hn_fac : IsUnit ((n - 1)! : A))
```

Comment:
And similarly for all the others
```suggestion
theorem IsUnit.natCast_factorial_of_lt {n : ℕ} (hn_fac : IsUnit ((n - 1)! : A))
```

Reply:
I put everything inside the `IsUnit` namespace.